TransformAST.cpp

```diff
@@ -0,0 +1,142 @@
+#include <string>
+
+#include <swift/AST/ASTContext.h>
+#include <swift/AST/ASTWalker.h>
+#include <swift/AST/Expr.h>
+#include <swift/AST/ParameterList.h>
+#include <swift/AST/Pattern.h>
+#include <swift/AST/Stmt.h>
+
+#include "TransformAST.h"
+
+void TransformFinalExpressionAndAddGlobal(swift::SourceFile &src_file)
+{
+    if(!src_file.Decls.empty())
+    {
+        swift::Decl *last_decl = *(src_file.Decls.end() - 1);
+        if(auto last_top_level_code_decl = llvm::dyn_cast<swift::TopLevelCodeDecl>(last_decl))
+        {
+            swift::ASTContext &ast_ctx = src_file.getASTContext();
+            llvm::MutableArrayRef<swift::ASTNode>::iterator back_iterator =
+                last_top_level_code_decl->getBody()->getElements().end() - 1;
+            swift::Expr *last_expr = (*back_iterator).dyn_cast<swift::Expr *>();
```

Comment:
Is this llvm::dyn_cast? Why not something like `llvm::dyn_cast<swift::Expr *>(*back_iterator)`?

Reply:
It gives some compiler error. This part was mostly copied from LLDB so I just did it their way:
```
S:\llvm\include\llvm/Support/Casting.h(329,10):  error: non-const lvalue reference to type 'swift::Expr *' cannot bind to a temporary of type 'swift::Expr *'
  return isa<X>(Val) ? cast<X>(Val) : nullptr;
         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
..\TransformAST.cpp(24,40):  note: in instantiation of function template specialization 'llvm::dyn_cast<swift::Expr *, swift::ASTNode>' requested here
        swift::Expr *last_expr = llvm::dyn_cast<swift::Expr *>(*back_iterator);
```